devices/linux/Files/main.py

```diff
@@ -125,48 +146,54 @@ class Config:
     init_info = ""
     init_confirmation = ""
     tou = ""
-    sb_user_file =  ""
+    sb_user_file = ""
     verify_user_realm_input = False
-    user_realm =  ""
+    user_realm = ""
     hint_user_input = False
 
 
-
 class InstallerData:
+    """ installer data """
     graphics = ''
+
     def __init__(self):
+        self.password = ''
+        self.username = ''
+
         self.__get_graphics_support()
-        self.show_info(Config.init_info.format(Config.instname, Config.email, Config.url))
-        if self.ask(Config.init_confirmation.format(Config.instname, Config.profilename), Messages.cont, 1):
+        self.show_info(Config.init_info.format(Config.instname,
+                                               Config.email, Config.url))
+        if self.ask(Config.init_confirmation.format(Config.instname,
+                                                    Config.profilename),
+                    Messages.cont, 1):
             sys.exit(1)
-        if Config.tou != None:
+        if Config.tou is not None:
             if self.ask(Config.tou, Messages.cont, 1):
                 sys.exit(1)
-        if os.path.exists(os.environ.get('HOME') + '/.cat_installer'):
-            if self.ask(Messages.cat_dir_exists.format(os.environ.get('HOME') + '/.cat_installer'), Messages.cont, 1):
+        if os.path.exists('{}/.cat_installer'.format(os.environ.get('HOME'))):
+            if self.ask(Messages.cat_dir_exists.format(
+                            os.environ.get('HOME') + '/.cat_installer'),
+                        Messages.cont, 1):
                 sys.exit(1)
         else:
-            os.mkdir(os.environ.get('HOME') + '/.cat_installer', 0o700)
-        certfile = os.environ.get('HOME') + '/.cat_installer/ca.pem'
-        with open(certfile, 'w') as f:
+            os.mkdir('{}/.cat_installer'.format(os.environ.get('HOME')), 0o700)
+        cert_file = '{}/.cat_installer/ca.pem'.format(os.environ.get('HOME'))
+        with open(cert_file, 'w') as f:
             f.write(Config.CA + "\n")
-        f.closed
 
-    def ask(self, question, prompt = '', default = None):
+    def ask(self, question, prompt='', default=None):
+        """  """
         if self.graphics == 'tty':
             yes = Messages.yes[:1].upper()
             no = Messages.no[:1].upper()
-            print("\n-------\n" + question + "\n")
+            print('\n-------\n{}\n'.format(question))
             while True:
-                p = prompt + " (" + Messages.yes + "/" + Messages.no + ") "
+                p = '{0} ({1}/{2}) '.format(prompt, Messages.yes, Messages.no)
                 if default == 1:
                     p += "[" + yes + "]"
                 elif default == 0:
-                    p += "[" + no + "]"  
-                try:
-                    inp = raw_input(p)
-                except:
-                    inp = input(p)
+                    p += "[" + no + "]"
+                inp = input(p)
```

Comment:
When we leave out `raw_input` (and by this make it incompatible to py2) we could as well change the shebang to python3 and remove the version-detection at top. `python3` *should* be the name of the py3 interpreter: https://www.python.org/dev/peps/pep-0394/ . I don't think we will meet systems at the intended audience that won't have a py3 environment. But some help to install py3-dbus would be good.

Reply:
In my opinion we shouldn't support ``python2`` anymore. The official support ends up with the 31.12.2019, so it is still for only one year!  
All normal distributions use python3 and we spend a lot less time to support. It is better if we improve the code further e.g. tests and documentation.

 And so on, it's not a big deal for a linux user to install ``python3``, if it hasn't yet.

Reply:
I think so, too. But in this case we should tell the user -- in GUI. But we should wait for @twoln's opinion before going forward with this.

Reply:
Yes, good idea. Should we add a issuie with all collected improvements?

Reply:
Done. See #116

Reply:
Thanks. Just for future reference: it's #120, not #116  ;).